### DISCUSSION TITLE
XML File Generation for a Waypoint Defined Trajectory

### DISCUSSION THREAD

**Discussion by cjdupuis1** [JSBSim-Team/jsbsim#686] [Q&A]
I am working on a project which autogenerate JSB Sim XML scripts to simulate airplane trajectories that the user can define via a waypoint system and also commands like turn left 90 degrees after 20 seconds, etc. (JSB Sim trajectories have very low jerks and that is important for my application). I currently have the system implemented for the c310 model since it already had the autopilots tuned and created.

Have you helped someone before on a similar project? (I have found mentions of similar projects on the web like the one at this link. There are mentions of a similar project at the bottom of this post https://github.com/JSBSim-Team/jsbsim/discussions/444).

I have some questions and a similar project with some available code and resources would be helpful

**Comment by jonyMarino** [comment]
Hi @cjdupuis1! I would like to see your project if possible. I would be glad to help if possible.

**Reply by cjdupuis1** [reply]
Hello, I don't have access to the code I was working on. So, can't help you there.
My suggestion would be to look at the waypoint example that is provided in the python JSBSim download.

**Comment by MariaALJ** [comment]
Hello, Im very interested in this subject, I've previously asked a similar question #747 about how to model a waypoint navigation system in unreal engine - JSBSim. So if you have any leads to share, I'd be interested. Thank you in advance!

**Reply by cjdupuis1** [reply]
Can't help with the unreal engine part, but building a waypoint system in JSB Sim is doable.
